src/meta/src/manager/cluster.rs

```diff
@@ -117,55 +117,57 @@ where
                     .unwrap_or_default();
             }
 
-            worker.update_ttl(self.max_heartbeat_interval);
             let current_parallelism = worker.worker_node.parallel_units.len();
             if current_parallelism == worker_node_parallelism
                 && worker.worker_node.property == property
             {
+                worker.update_ttl(self.max_heartbeat_interval);
```

Comment:
This is actually a separated issue: TTL should not be updated on failed registration.